packages/react-composites/src/composites/CallComposite/components/buttons/Custom.tsx

```diff
@@ -1,47 +1,100 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { ControlBarButton } from '@internal/react-components';
+import { Icon } from '@fluentui/react';
+import { ControlBarButton, ControlBarButtonStrings } from '@internal/react-components';
 import React from 'react';
 import {
   CallControlDisplayType,
   CustomCallControlButtonCallback,
   CustomCallControlButtonPlacement
 } from '../../types/CallControlOptions';
 
+/**
+ * Max number of Custom Buttons in primary and secondary ControlBar
+ * Does not include existing buttons in the controlBar.
+ *
+ * @private
+ */
+export const CUSTOM_BUTTON_OPTIONS = {
+  MAX_PRIMARY_DESKTOP_CUSTOM_BUTTONS: 3,
+  MAX_PRIMARY_MOBILE_CUSTOM_BUTTONS: 1,
+  MAX_SECONDARY_DESKTOP_CUSTOM_BUTTONS: 2
+};
+
 /** @private */
 export type CustomButtons = { [key in CustomCallControlButtonPlacement]: JSX.Element | undefined };
 
 /** @private */
-export const generateCustomButtons = (
+export const generateCustomControlBarButtons = (
   onFetchCustomButtonProps?: CustomCallControlButtonCallback[],
   displayType?: CallControlDisplayType
 ): CustomButtons => {
   const response = {
-    first: undefined,
-    afterCameraButton: undefined,
-    afterEndCallButton: undefined,
-    afterMicrophoneButton: undefined,
-    afterDevicesButton: undefined,
-    afterParticipantsButton: undefined,
-    afterScreenShareButton: undefined,
-    last: undefined
+    primary: undefined,
+    secondary: undefined,
+    overflow: undefined
   };
   if (!onFetchCustomButtonProps) {
     return response;
   }
 
+  const generateCustomControlBarButtonStrings = (text: string | undefined): ControlBarButtonStrings => {
+    return { label: text };
+  };
+
   const allButtonProps = onFetchCustomButtonProps.map((f) => f({ displayType }));
   for (const key in response) {
     response[key] = (
       <>
         {allButtonProps
           .filter((buttonProps) => buttonProps.placement === key)
           .map((buttonProps, i) => (
-            <ControlBarButton {...buttonProps} key={`${buttonProps.placement}_${i}`} />
+            <ControlBarButton
+              onClick={buttonProps.onItemClick}
+              showLabel={buttonProps.showLabel}
+              strings={generateCustomControlBarButtonStrings(buttonProps.text)}
+              styles={buttonProps.styles}
+              key={`${buttonProps.placement}_${i}`}
+              onRenderIcon={() => <Icon iconName={buttonProps.icon} />}
```

Comment:
should this be? `<CallCompositeIcon iconName={buttonProps.icon} />`

Reply:
Good catch, yes. This would be a good argument for separating custom.tsx into two files, one for Call Composite and one for CallWithChat Composite.